Sources/Kitura/String+Extensions.swift

```diff
@@ -24,9 +24,11 @@ extension String {
     /// values.
     var urlDecodedFieldValuePairs: [String: String] {
         var result: [String: String] = [:]
-        for item in self.components(separatedBy: "&") {
-            let (key, value) = item.keyAndDecodedValue
-            if let value = value {
+        for item in self.split(separator: "&") {
+            let (keySub, valueSub) = item.keyAndDecodedValue
```

Comment:
Why not make `keyAndDecodedValue` continue to return `(String, String?)`? There must be a reason you changed it but I couldn't spot it...

Reply:
Oh is it because `item` is a substring? If so, why not make it a string before calling `keyAndDecodedValue`?

Reply:
This was to save cost, there's no need to may the key a String if the value is nil.